### PR TITLE
clang-tidy fixups

### PR DESCRIPTION
**Pull request by neheb**
## Description

clang-tidy fixups.

### CODE REVIEW

File: src/iothread.cpp [hunk 1]
```diff
@@ -199,7 +199,9 @@ static void enqueue_thread_result(work_request_t req) {
     s_result_queue.acquire()->push(std::move(req));
 }
 
-static void *this_thread() { return (void *)(intptr_t)pthread_self(); }
+static void *this_thread() {
+    return reinterpret_cast<void *>(static_cast<intptr_t>(pthread_self()));
```
Comment (faho) [inline src/iothread.cpp]: Breaks the build on macOS:

>/Users/travis/build/fish-shell/fish-shell/src/iothread.cpp:203:37: error: static_cast from 'pthread_t _Nonnull' (aka '_opaque_pthread_t *') to 'intptr_t' (aka 'long') is not allowed

>return reinterpret_cast<void *>(static_cast<intptr_t>(pthread_self()));

File: src/complete.cpp [hunk 2]
```diff
@@ -214,7 +214,7 @@ completion_t::completion_t(wcstring comp, wcstring desc, string_fuzzy_match_t ma
       flags(resolve_auto_space(completion, flags_val)) {}
 
 completion_t::completion_t(const completion_t &) = default;
-completion_t::completion_t(completion_t &&) = default;
+completion_t::completion_t(completion_t &&) noexcept = default;
```
Comment (faho) [inline src/complete.cpp]: Breaks the build on Travis on Linux:

>The noexcept thing breaks the build on Travis, and it's useless as we build without exceptions anyway:

>/home/travis/build/fish-shell/fish-shell/src/complete.cpp:219:15: error: function ‘completion_t& completion_t::operator=(completion_t&&)’ defaulted on its redeclaration with an exception-specification that differs from the implicit exception-specification ‘’

Comment (neheb) [reply src/complete.cpp]: Fixed. Probably older GCC.

File: src/iothread.cpp (hunk 1 above)
Comment (neheb) [reply src/iothread.cpp]: Switched back to C cast for that since it's static on some platforms and reinterpret_cast on others.

**Comment by neheb** [comment]
Fixed.

File: src/screen.cpp [hunk 3]
```diff
@@ -1153,7 +1153,7 @@ void s_reset(screen_t *s, screen_reset_mode_t mode) {
         if (screen_width > non_space_width) {
             bool justgrey = true;
             if (cur_term && enter_dim_mode) {
-                std::string dim = tparm((char *)enter_dim_mode);
+                std::string dim = tparm(enter_dim_mode);
```
Comment (faho) [inline src/screen.cpp]: This cast is necessary on some platforms. Specifically I'm pretty sure this breaks the build on NetBSD.

File: src/screen.cpp [hunk 4]
```diff
@@ -1164,22 +1164,22 @@ void s_reset(screen_t *s, screen_reset_mode_t mode) {
             if (cur_term && justgrey && set_a_foreground) {
                 if (max_colors >= 238) {
                     // draw the string in a particular grey
-                    abandon_line_string.append(str2wcstring(tparm((char *)set_a_foreground, 237)));
+                    abandon_line_string.append(str2wcstring(tparm(set_a_foreground, 237)));
```
Comment (faho) [inline src/screen.cpp]: These casts.

**Comment by floam** [comment]
I don't like the "auto all the things" commit.

**Comment by faho** [comment]
Yeah, I'm gonna merge this partially. I'm dropping:

- "use auto"
- "apply performance changes" because it adds the "noexcept" to the completion_t constructor that breaks the build
- "replace C casting with C++", because it breaks tparm and looks awful
- "mark move constructors noexcept" because we do not use exceptions

To be honest, I think we've reached the limits of what these static tools can help us with. Most changes they now propose either break something or don't improve things (just c++-ification for the sake of it).

So: Thank you, that'll be enough for now!

Merged as b26637042..a410dcb5c

**Comment by neheb** [comment]
Hardly. This is only version 9 of LLVM. GCC's also working on its own static analyzer.

**Pull request closed without merging**